deploy_stack.ps1

```diff
@@ -25,9 +25,7 @@ if (Get-Command docker -errorAction SilentlyContinue)
         throw "Docker not in swarm mode, please initialise the cluster (`docker swarm init`) and retry"
     }
 
-    # AE: would be nice to avoid this dependency.
-    Add-Type -AssemblyName System.Web
-    $password = [System.Web.Security.Membership]::GeneratePassword(24,5)
+    $password = -join ((33..126) * 120 | Get-Random -Count 24 | ForEach-Object { [char]$_ })
```

Comment:
I am not keen on this approach vs the original Web security API call.

Can you look at whether there is an equivalent in .NET Core that can be used?

Reply:
I agree, however, there does not seem to be a comparable alternative built in. Based on the article [here](https://poshhelp.wordpress.com/2017/01/30/why-you-should-stop-using-generatepassword/) it looks like the `GeneratePassword` function may resemble the following: 

```powershell
Function GeneratePassword ([int]$Length) {
    Add-Type -AssemblyName System.Web
    $CharSet = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789{]+-[*=@:)}$^%;(_!&#?>/|.'.ToCharArray()
    #Index1s 012345678901234567890123456789012345678901234567890123456789012345678901234567890123456
    #Index10s 0 1 2 3 4 5 6 7 8
 
    $rng = New-Object System.Security.Cryptography.RNGCryptoServiceProvider
    $bytes = New-Object byte[]($Length)
 
    $rng.GetBytes($bytes)
 
    $Return = New-Object char[]($Length)
 
    For ($i = 0 ; $i -lt $Length ; $i++) {
        $Return[$i] = $CharSet[$bytes[$i]%$CharSet.Length]
    }
 
    Return (-join $Return)
 
}
```

The result would resemble:

```powershell
GeneratePassword -Length 24

$@Unw_C6xQ+W*YBo#HEFBJv2
```

Would you be more comfortable adding this function to the deploy script?

Reply:
Any thoughts @alexellis ?

Reply:
Yes, I'm happier with the original suggestion in the linked issue under `generatePassword`

Is there any reason why you are using Swarm and not Kubernetes?

You can also use Git Bash to get around this on any Windows system.

Reply:
@alexellis ,

If you would like to support PowerShell core then I think the only option is the function I referenced [here](https://github.com/openfaas/faas/pull/1476#discussion_r396425982). I believe it to be just as secure as the original .Net class that is being used. 

I am currently using Swarm as that is what we have deployed.

Sure, there are other workarounds, but I am simply trying to address the situation where PowerShell Core may be used (Windows, Linux, OSX). Since there is already support for some PowerShell versions I wanted to expand on that. If you don't agree or find it beneficial then I will discontinue working on a solution.